app/src/main/java/at/aau/edu/wizards/gameModel/GameModel.kt

```diff
@@ -4,14 +4,16 @@ class GameModel {
 
     private val players = ArrayList<GameModelPlayer>()
     private var dealer = GameModelDealer(0)
-    private var rules = GameModelRules(players, 0, dealer, this)
+    private var rules = GameModelRules(players, 0, dealer, this, 0)
     var listener = GameModelListener(rules, players)
         private set
 
     fun sendMessage(move: String): Boolean {
         if ((legalMessageGuess(move) && (move[0].code - 60) / 11 == rules.id) || (legalMessageCard(
                 move
-            ) && rules.localPlayerOwns(dealer.getCardFromHash(move[0].code))) || move == "EndGame"
+            ) && rules.localPlayerOwns(dealer.getCardFromHash(move[0].code)) && (rules.winningCard !is GameModelCard.Normal || ((rules.winningCard as GameModelCard.Normal).color == getColor(
```

Comment:
please simplify

Reply:
Split into multiple functions and tried to give describing names. Code should be quite clear if you look at it now in context.